File: src/lib/secrets.ts
import { invoke } from '@tauri-apps/api/core';

/**
 * 敏感信息统一存入系统钥匙串（Windows Credential Manager / macOS Keychain / Linux Secret Service），
 * localStorage 只保存非敏感配置。所有操作均为异步。
 */
const PREFIX = 'spurh';

export function secretKey(name: string): string {
  return PREFIX + '.' + name;
}

export async function setSecret(name: string, value: string): Promise<void> {
  if (!value) {
    await deleteSecret(name);
    return;
  }
  await invoke('secret_set', { key: secretKey(name), value });
}

export async function getSecret(name: string): Promise<string | null> {
  return invoke<string | null>('secret_get', { key: secretKey(name) });
}

export async function deleteSecret(name: string): Promise<void> {
  await invoke('secret_delete', { key: secretKey(name) }).catch(() => undefined);
}

File: src-tauri/src/secrets.rs
// 系统钥匙串：AI API Key、数据库密码、SSH 密码等敏感信息不再落盘到 localStorage。
// Windows 使用 Credential Manager，macOS 使用 Keychain，Linux 使用 Secret Service。
use keyring::Entry;

const SERVICE: &str = "dev.spurh.app";

fn entry(key: &str) -> Result<Entry, String> {
    Entry::new(SERVICE, key).map_err(|error| format!("无法访问系统钥匙串：{error}"))
}

#[tauri::command]
pub fn secret_set(key: String, value: String) -> Result<(), String> {
    entry(&key)?.set_password(&value).map_err(|error| format!("写入钥匙串失败：{error}"))
}

#[tauri::command]
pub fn secret_get(key: String) -> Result<Option<String>, String> {
    match entry(&key)?.get_password() {
        Ok(value) => Ok(Some(value)),
        Err(keyring::Error::NoEntry) => Ok(None),
        Err(error) => Err(format!("读取钥匙串失败：{error}")),
    }
}

#[tauri::command]
pub fn secret_delete(key: String) -> Result<(), String> {
    match entry(&key)?.delete_credential() {
        Ok(()) => Ok(()),
        Err(keyring::Error::NoEntry) => Ok(()),
        Err(error) => Err(format!("删除钥匙串条目失败：{error}")),
    }
}
